packages/block-library/src/buttons/edit.native.js

```diff
@@ -0,0 +1,144 @@
+
+	useEffect( () => {
+		const margins = 2 * styles.parent.marginRight;
+		const { width } = sizes || {};
+		if ( width ) {
+			setMaxWidth( width - margins );
+		}
+	}, [ sizes ] );
+
+	function renderAppender() {
+		if ( shouldRenderFooterAppender ) {
+			return (
+				<View style={ styles.appenderContainer }>
+					<InnerBlocks.ButtonBlockAppender
+						isFlex={ false }
+						onAddBlock={ onAddNextButton }
+					/>
+				</View>
+			);
+		}
+		return null;
+	}
+
+	// Inside buttons block alignment options are not supported.
+	const alignmentHooksSetting = {
+		isEmbedButton: true,
+	};
+
+	return (
+		<AlignmentHookSettingsProvider value={ alignmentHooksSetting }>
+			{ resizeObserver }
+			<InnerBlocks
+				allowedBlocks={ ALLOWED_BLOCKS }
+				template={ BUTTONS_TEMPLATE }
+				renderAppender={ renderAppender }
+				__experimentalMoverDirection="horizontal"
+				horizontalAlignment={ align }
+				onDeleteBlock={ shouldDelete && onDelete }
+				onAddBlock={ onAddNextButton }
+				parentWidth={ maxWidth }
+				shouldRenderFooterAppender={ shouldRenderFooterAppender }
+				marginHorizontal={ spacing }
+				marginVertical={ spacing }
+			/>
+		</AlignmentHookSettingsProvider>
+	);
+}
+
+export default compose(
+	withSelect( ( select, { clientId } ) => {
+		const {
+			getBlockCount,
+			getBlockParents,
+			getSelectedBlockClientId,
+		} = select( 'core/block-editor' );
+		const selectedBlockClientId = getSelectedBlockClientId();
+		const selectedBlockParents = getBlockParents(
+			selectedBlockClientId,
+			true
+		);
+
+		return {
+			// The purpose of `shouldDelete` check is giving the ability to pass to
+			// mobile toolbar function called `onDelete` which removes the whole
+			// `Buttons` container along with the last inner button when
+			// there is exactly one button.
+			shouldDelete: getBlockCount( clientId ) === 1,
+			isInnerButtonSelected: selectedBlockParents[ 0 ] === clientId,
+		};
+	} ),
+	withDispatch( ( dispatch, { clientId }, registry ) => {
+		const { replaceInnerBlocks, selectBlock, removeBlock } = dispatch(
+			'core/block-editor'
+		);
+		const { getBlocks, getBlockOrder } = registry.select(
+			'core/block-editor'
+		);
+		const innerBlocks = getBlocks( clientId );
+
+		return {
+			// The purpose of `onAddNextButton` is giving the ability to automatically
+			// adding `Button` inside `Buttons` block on the appender press event.
+			onAddNextButton: ( selectedId ) => {
+				const order = getBlockOrder( clientId );
+				const selectedButtonIndex = order.findIndex(
+					( i ) => i === selectedId
+				);
+
+				const index =
+					selectedButtonIndex === -1
+						? order.length + 1
+						: selectedButtonIndex;
+
+				const insertedBlock = createBlock( 'core/button' );
+
+				innerBlocks.splice( index + 1, 0, insertedBlock );
+
+				replaceInnerBlocks( clientId, innerBlocks, true );
+				selectBlock( insertedBlock.clientId );
+			},
+			onDelete: () => {
+				removeBlock( clientId );
+			},
```

Comment:
This `onDelete` is passed thought InnerBlocks and gets to remove button in MobileToolbar for each Button children.

Please see comment above in Button because I think we can also pass it to BlockEdit in `block.navite.js`. Then we can skip create another `onDelete` action in Button